nntrainer/models/neuralnet.cpp

```diff
@@ -358,15 +363,23 @@ void NeuralNetwork::saveModel() {
 void NeuralNetwork::readModel() {
   if (!isFileExist(save_path))
     return;
+
+  NeuralNetwork tmp(*this);
+
   std::ifstream model_file(save_path, std::ios::in | std::ios::binary);
   for (unsigned int i = 0; i < layers.size(); i++)
-    layers[i]->read(model_file);
+    tmp.layers[i]->read(model_file);
   if (continue_train) {
-    model_file.read((char *)&epoch_idx, sizeof(epoch_idx));
-    model_file.read((char *)&iter, sizeof(iter));
+    checkedRead(model_file, (char *)&tmp.epoch_idx, sizeof(epoch_idx),
```

Comment:
Even without continue_train, this value must be read. Otherwise the values read after this will be wrong.

Reply:
Good point. I'll fix it

Reply:
Done